src/network/receiver.go

```diff
@@ -0,0 +1,57 @@
+package network
+
+import (
+	"net"
+	"strconv"
+	"time"
+)
+
+/*
+Listen for incoming messages on specified IP and port.
+*/
+func ListenForMessages(ip string, port int, messageChannelchan chan<- []byte) {
+	conn, err := net.ListenPacket("udp4", ip+":"+strconv.Itoa(port))
```

Comment:
For concatenating strings I have used fmt.Sprintf, we need to agree on one of the methods to be concise. You will have to change from `net.ListenPacket` to `reuseport.ListenPacket` in order to bind to same port on multiple processes on same machine.